src/Gui/Windows/CombinedCodeViewInteractor.cs

```diff
@@ -299,12 +304,39 @@ public bool ChooseTextEncoding()
             return true;
         }
 
-        private void MixedCodeDataView_MouseDown(object sender, EventArgs e)
+        private void MixedCodeDataView_MouseUp(object sender, MouseEventArgs e)
+        {
```

Comment:
I'd be happier with this change if it occurred after an explicit user action, rather than always on left-mouse-up. Consider defining a context menu for the MixedCodeDataView, and when the user selects the context menu, display the declarationTextBox. See `LowLevelnteractor.MarkType()` for an example. You can even define a shortcut key (`Ctrl+T` is used by the `LowLevelInteractor` to mark types.

In fact, as I write this, it makes total sense to integrate TypeMarker and DeclarationTextBox; they're almost doing the same thing, except that TypeMarker is using a dialect of "[Hungarian](https://en.wikipedia.org/wiki/Hungarian_notation)" for speed and convenience.